src/parsing/Parse_rule.mli

```diff
@@ -1,15 +1,33 @@
+(*
+   Rule parsing
+*)
+
+(*
+   For rules that come from a local file, this is the path to the file
+   containing the rule. It's used to create a dotted rule identifier
+   such as 'stuff.rules.foo' for a rule declared as 'foo' in the file
+   'stuff/rules.yml'.
```

Comment:
maybe add a comment like
"This may seem redundant with the second argument of parse_and_filter_invalid_rules() below, but sometimes
this second argument corresponds to a temporary file generated from content fetched from the network (e.g., rules coming from the registry locally stored in /tmp/) in which case we do not want to add this prefix, hence the use of Other_origin in that case, even if we're still parsing a local (temporary) file."